Makefile

```diff
@@ -0,0 +1,65 @@
+.PHONY: all help build run builddocker rundocker kill rm-image rm clean enter logs
+
+all: help
+
+help:
+	@echo ""
+	@echo "-- Help Menu"
+	@echo ""   1. make run       - build and run docker container
+	@echo ""   2. make build     - build docker container
+	@echo ""   3. make clean     - kill and remove docker container
+	@echo ""   4. make enter     - execute an interactive bash in docker container
+	@echo ""   3. make logs      - follow the logs of docker container
+
+build: NAME TAG builddocker
+
+# run a plain container
+run: PORT NAME TAG build rundocker
+
+rundocker:
+	$(eval TMP := $(shell mktemp -d --suffix=DOCKERTMP))
+	$(eval NAME := $(shell cat NAME))
+	$(eval TAG := $(shell cat TAG))
+	$(eval PORT := $(shell cat PORT))
+	chmod 777 $(TMP)
+	@docker run --name=$(NAME) \
+	--cidfile="cid" \
+	-v $(TMP):/tmp \
+	-d \
+	-p $(PORT):80 \
+	-t $(TAG)
+
+builddocker:
+	/usr/bin/time -v docker build -t `cat TAG` .
+
+kill:
+	-@docker kill `cat cid`
+
+rm-image:
+	-@docker rm `cat cid`
+	-@rm cid
+
+rm: kill rm-image
+
+clean: rm
+
+enter:
+	docker exec -i -t `cat cid` /bin/bash
+
+logs:
+	docker logs -f `cat cid`
+
+NAME:
+	@while [ -z "$$NAME" ]; do \
+		read -r -p "Enter the name you wish to associate with this container [NAME]: " NAME; echo "$$NAME">>NAME; cat NAME; \
+	done ;
+
+TAG:
+	@while [ -z "$$TAG" ]; do \
+		read -r -p "Enter the tag you wish to associate with this container [TAG]: " TAG; echo "$$TAG">>TAG; cat TAG; \
+	done ;
+
+PORT:
+	@while [ -z "$$PORT" ]; do \
+		read -r -p "Enter the tag you wish to associate with this container [PORT]: " PORT; echo "$$PORT">>PORT; cat PORT; \
```

Comment:
The string should be "Enter the **port** you wish to associate with this container [PORT]: "